core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -579,6 +579,8 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
           if (quota == null)
             quota = staticConfigClientIdQuota
         }
+        if (quota == null)
+          quota = ClientQuotaManagerConfig.UnlimitedQuota
```

Comment:
Rather than change the default callback, I think it may be better to change the caller `updateQuotaMetricConfigs` to handle `null` case better. At the moment, `updateQuotaMetricConfigs` updates metrics only if `quotaCallback.quotaLimit` is non-null. This works in most cases because `null` indicates that those tags are no longer in use and hence we don't need to update them - those metrics will eventually expire. 

The problem is once we enable any quotas, we always use quota metrics and the tags depend on what quotas were enabled. To be safe, we should update existing metrics regardless of whether the callback returns null. This PR addresses the issue for the default callback. But since this is a valid case for any callback implementation, using `ClientQuotaCallback.quotaLimit`(that returns a non-null value ) instead of `quotaCallback.quotaLimit` in `updateQuotaMetricConfigs` would ensure that we always update existing metrics with any callback.

Reply:
This makes sense. Thanks Rajini, will update my PR.